network/network_linux.go

```diff
@@ -232,22 +234,37 @@ func isGreaterOrEqaulUbuntuVersion(versionToMatch int) bool {
 		retrieved_version, err := getMajorVersion(version)
 		if err != nil {
 			log.Printf("[net] Not setting dns. Unable to retrieve major version: %v", err)
-			return false
+			return 0
 		}
 
-		if retrieved_version >= versionToMatch {
-			return true
-		}
+		return retrieved_version
+	}
+
+	return 0
+}
+
+func isGreaterOrEqaulUbuntuVersion(versionToMatch int) bool {
+	ubuntuVerion := getUbuntuVersion()
+	if ubuntuVerion >= versionToMatch {
+		return true
 	}
 
 	return false
 }
 
-func readDnsInfo(ifName string) (DNSInfo, error) {
-	var dnsInfo DNSInfo
+func readDNSInfo(ifName string, osVersion int) (DNSInfo, error) {
+	var (
+		dnsInfo DNSInfo
+		cmd     string
+	)
 
 	p := platform.NewExecClient()
-	cmd := fmt.Sprintf("systemd-resolve --status %s", ifName)
+	if osVersion >= ubuntuVersion22 {
+		cmd = fmt.Sprintf("resolvectl status %s", ifName)
+	} else {
+		cmd = fmt.Sprintf("systemd-resolve --status %s", ifName)
+	}
```

Comment:
to me it seems like the more reliable and generic way of accomplishing this would be to check for the presence/status of `resolvectl` and `systemd-resolved` and use the appropriate one based on that result.

Reply:
although if I'm reading the [ubuntu manpage](https://manpages.ubuntu.com/manpages/impish/man1/resolvectl.1.html) correctly, it seems like invoking `resolvectl` via `systemd-resolve` should be backwards compatible and I'm wondering why we even need this?

Reply:
Hi Evan, I mean "resolvectl" command is not backwards compatible with Ubuntu18 and I tried it on one Ubuntu18.04 machine:
![Image](https://github.com/Azure/azure-container-networking/assets/129891899/2ba83c09-fc37-4809-a248-bf1a5174df3c)
And "systemd-resolve" is not able to be used on Ubuntu22